File: app/src/main/java/com/example/andrey/teso/servers/ServerListAdapter.java
package com.example.andrey.teso.servers;


import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.example.andrey.teso.R;

import java.util.ArrayList;

public class ServerListAdapter extends ArrayAdapter<Server> {

    private Context mContext;
    private int mResource;
    private int lastPosition = -1;

    private static class ViewHolder {
        TextView idLocation;
        TextView distance;
    }

    public ServerListAdapter(Context context, int resource, ArrayList<Server> objects) {
        super(context, resource, objects);
        mContext = context;
        mResource = resource;
    }

    @NonNull
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        Integer id = getItem(position).getIdentifier();
        String location = getItem(position).getLocation();
        Integer distance = getItem(position).getDistance();

        Server server = new Server(id, location, distance);

        final View result;
        ViewHolder holder;

        if(convertView == null){
            LayoutInflater inflater = LayoutInflater.from(mContext);
            convertView = inflater.inflate(mResource, parent, false);
            holder= new ViewHolder();
            holder.idLocation = convertView.findViewById(R.id.textView1);
            holder.distance = convertView.findViewById(R.id.textView2);

            result = convertView;
            convertView.setTag(holder);
        }
        else{
            holder = (ViewHolder) convertView.getTag();
            result = convertView;
        }

        Animation animation = AnimationUtils.loadAnimation(mContext, (position > lastPosition) ? R.anim.load_down : R.anim.load_up);
        result.startAnimation(animation);
        lastPosition = position;

        holder.idLocation.setText(server.getLocation() + " #" + server.getIdentifier());
        holder.distance.setText(server.getDistance() + " km");

        return convertView;
    }
}

File: app/src/main/java/com/example/ievazygaite/androidparty/ui/list/ServerListAdapter.kt
package com.example.ievazygaite.androidparty.ui.list

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.ievazygaite.androidparty.R
import com.example.ievazygaite.androidparty.data.server.Server
import kotlinx.android.synthetic.main.item_layout.view.*

class ServerListAdapter(private val serverList: ArrayList<Server>, private val context: Context) :
    RecyclerView.Adapter<ServerListAdapter.OrderViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, position: Int): OrderViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_layout, parent, false)
        return OrderViewHolder(view)
    }

    override fun onBindViewHolder(holder: OrderViewHolder, position: Int) {
        holder.bind(serverList[position])
    }

    override fun getItemCount(): Int = serverList.size

    fun setData(servers: List<Server>) {
        serverList.clear()
        serverList.addAll(servers)
        notifyDataSetChanged()
    }

    inner class OrderViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(server: Server) {
            itemView.distance.text = context.getString(R.string.distance_in_km, server.distance)
            itemView.server_name.text = server.serverName
        }
    }

}